### ISSUE TITLE
Reduce LR on plateau but not increase

### ISSUE THREAD

**Issue by peastman** [mir-group/nequip#326]
The ReduceLROnPlateau option actually looks for the loss to increase, not plateau.  As long as it doesn't increase, the learning rate doesn't change.

In training my model I never see the loss increase.  It just keeps decreasing by tinier and tinier amounts.  Could we add a margin to the test, so for example I could tell it to reduce the learning rate any time the loss decreases by less than 2%?

**Comment by Linux-cpp-lisp** [comment]
I think there is an option for this in PyTorch, though I ahven't looked super closely. See `is_better` in https://pytorch.org/docs/1.11/_modules/torch/optim/lr_scheduler.html#ReduceLROnPlateau.

You can specify these options in the YAML with `lr_scheduler_*` keys.

**Comment by peastman** [comment]
That's exactly what I need.  These options aren't listed in https://github.com/mir-group/nequip/blob/main/configs/full.yaml.  Do we need code changes to support them?  Or will it automatically translate any `lr_scheduler_X` option into the `X` argument of the scheduler?

**Comment by simonbatzner** [comment]
This should already be implemented, you can set a delta on the improvement, see [here](https://github.com/mir-group/nequip/blob/main/configs/full.yaml#L197) and [here](https://github.com/mir-group/nequip/blob/main/nequip/train/early_stopping.py#L32). Let us know if you have questions re this.

**Comment by peastman** [comment]
> This should already be implemented, you can set a delta on the improvement, see [here](https://github.com/mir-group/nequip/blob/main/configs/full.yaml#L197) and [here](https://github.com/mir-group/nequip/blob/main/nequip/train/early_stopping.py#L32).

That refers to early stopping, not to reducing the learning rate.

**Comment by simonbatzner** [comment]
Oh my bad, of course, misread.

**Comment by peastman** [comment]
Setting `lr_scheduler_threshold` works perfectly.  Thanks!  Can we document in `configs/full.yaml` that you can specify any argument to the scheduler?

**Comment by Linux-cpp-lisp** [comment]
Done 👍

**Comment by Linux-cpp-lisp** [comment]
Done 👍
